### DISCUSSION TITLE
Don't punish non-elite ISP connections.

### DISCUSSION THREAD

**Discussion by Layarion** [FakeFishGames/Barotrauma#5276] [Ideas and suggestions]
Me and my brother like to play games together and have a co-op experience, but we've had a hard time doing that on this game. We have rather imperfect internet connections, and lose internet at one house or another.

When our internet goes out, it often goes out for 3-5 minutes at a time. my connection is actually worse - it goes out for 3-5 minutes, but can happen many times a day at random.

Anyway we had to stop playing your game together because you punish us heavily each time it happens. The current mechanics 1) Mean to reconnect we gotta increase the DC time in the config file, but that doesn't stop us from drowning. 2) If we don't do that, we still have to restart the entire mission. 3) The commands available for reviving such a dead playing marks the campaign as cheating - and so we can't get achievements. 4) we could go in the config file and change it afterward - but this still feels like a punishment to legit players with bad connections.

We both live in the same state and country, USA - and I'm wired up (not wifi) and have had my ISP out here many times. The connection goes out on my ISP side, at the modem level. So it's not an issue with my computer, but between my modem and ISP.

To re-summarize, 1) it feels punishing to have ISP's that aren't the best. We stopped playing this together after a few missions. 2) my internet can and has gone out at random - sometimes 4 times in half an hour, other times i can get a solid 12 hours with no issue. I have no ability to predict which it'll be though, and it's ruined our experience on this game. The upcoming respawn mechanic for campaigns would only help if my ISP went out less often, but it doesn't. I'm sure i'm not the only gamer here that has a "non-elite ISP".

**Comment by diegoyocys** [comment]
seems like a problem with your internet supplier, not the game. 

On top of that, most games (if not all) that do not have respawn options, multiplayer is NOT designed to lose connection and come back 5 mins later.

**Reply by Veritius** [reply]
Notably Barotrauma is already very lenient with this system, letting you repossess your character in the event of an accidental disconnect (but this is only available for a few minutes). Some games just kill you on the spot.

**Comment by Layarion** [comment]
That's not true at all, i've played games that aren't punishing on disconnecting players.

also, "problem with your internet" if that's the approach we want to take, then the devs shouldn't help with griefers or game crashing. the one is a problem with the way you find people to play with, and the other is a problem with the hardware or software you play on because perhaps this one craash doesn't happen on their machines.

these just seem like invalid reasons to ignore the possibility to do more, and i already said that as it stands - it's not enough. They need to at the very least - make the command to reconnect and revive dc'ed players not trigger a cheater flag or remove steam achievements. irrc the last time I used it I think we had to use 2 separate commands to get it working rightly like nothing happened. So they could streamline this into one command with a flag to revive if necessary.

last, i actually forgot to mention something important - sometimes the system fails to let us reconnect if the host had a long time offline, like the 5 -10 minutes i sometimes get stuck with. i think the issue here is that the server gets removed from the steam list, and i've had trouble with my brother reconnecting. at least, that's a guess as to why he sometimes had trouble reconnecting.

so to resummarize:
1) the commands need to not disable steam achievements
2) it should be rolled into one command that's more intuitive, because irrc it was something unintuitive live LiveActor something something command.
3) your arguments about "but other games don't do it" is only true for some games, and 3.1) doesn't even mean you shouldn't do it.

**Comment by Krzeszny** [comment]
Solution: make AI start controlling characters of players who lose connection.
